server/src/main/java/org/elasticsearch/common/util/concurrent/EsThreadPoolExecutor.java

```diff
@@ -66,9 +70,19 @@ public class EsThreadPoolExecutor extends ThreadPoolExecutor {
         this.contextHolder = contextHolder;
     }
 
+    @Override
+    public void setCorePoolSize(int corePoolSize) {
+        throw new UnsupportedOperationException("reconfiguration at runtime is not supported");
+    }
+
+    @Override
+    public void setMaximumPoolSize(int maximumPoolSize) {
+        throw new UnsupportedOperationException("reconfiguration at runtime is not supported");
+    }
+
     @Override
     public void execute(Runnable command) {
-        final Runnable wrappedRunnable = wrapRunnable(command);
+        final Runnable wrappedRunnable = command != WORKER_PROBE ? wrapRunnable(command) : WORKER_PROBE;
```

Comment:
Would it be bad to just return here if the command is `WORKER_PROBE`? (I did not check if super does bookkeeping that we must do anyway)

Reply:
`super.execute` is required, otherwise the probe is useless

Reply:
Oh yes, silly me, we _need_ the side effects!